core/src/main/java/org/mule/runtime/core/internal/event/AbstractEventContext.java

```diff
@@ -74,7 +74,7 @@
   private volatile byte state = STATE_READY;
   private volatile Either<Throwable, CoreEvent> result;
 
-  private final Set<ResponsePublisher> responsePublishers = new HashSet<>();
+  private LazyValue<ResponsePublisher> responsePublisher = new LazyValue<>(() -> new ResponsePublisher());
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Replace this lambda with a method reference. ([squid:S1612](https://sonarqube.kbuild.msap.io/coding_rules#rule_key=squid%3AS1612))

[See it in SonarQube](https://sonarqube.kbuild.msap.io/project/issues?id=mule-runtime-mule&pullRequest=7701&issues=AWkLOTwXotIXnmsUkh1-&open=AWkLOTwXotIXnmsUkh1-)

[ISSUE_KEY]: AWkLOTwXotIXnmsUkh1-